test/unit/client_tests.rb

```diff
@@ -105,6 +105,24 @@ class MixinTests < UnitTests
       assert_equal({}, call.job.params['event_params'])
     end
 
+    should "enqueue a dispatch job with a custom publisher using `publish_as`" do
+      publisher = Factory.string
+      channel   = Factory.string
+      name      = Factory.string
+      params    = @job_params
+      result = subject.publish_as(publisher, channel, name, params)
+
+      call = subject.enqueue_calls.last
+      dispatch_job = Factory.dispatch_job({
+        :event_channel   => channel,
+        :event_name      => name,
+        :event_params    => params,
+        :event_publisher => publisher
+      })
+      assert_equal dispatch_job.name,   call.job.name
+      assert_equal dispatch_job.params, call.job.params
+    end
```

Comment:
@jcredding I'd add tests for returning the event and also enqueueing on the dispatch queue like you do in the publish method tests above.